cmd/solo-kit-gen/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/solo-io/solo-kit/pkg/code-generator/codegen"
+	"github.com/solo-io/solo-kit/pkg/code-generator/docgen"
+	"github.com/solo-io/solo-kit/pkg/code-generator/model"
+	"github.com/solo-io/solo-kit/pkg/code-generator/parser"
+	"github.com/solo-io/solo-kit/pkg/errors"
+	"github.com/solo-io/solo-kit/pkg/utils/log"
+)
+
+func Gopath() string {
+	return os.Getenv("GOPATH")
+}
+
+var commonImports = []string{
+	"-I" + Gopath() + "/src",
+	"-I" + Gopath() + "/src/github.com/solo-io/solo-kit/api/external",
+}
+
+func main() {
+	if err := Main(); err != nil {
+		log.Fatalf("%v", err)
+	}
+}
+
+//
+//type arrayFlags []string
+//
+//func (i *arrayFlags) String() string {
+//	if i == nil {
+//		return "nil"
+//	}
+//	return fmt.Sprintf("%v", *i)
+//}
+//
+//func (i *arrayFlags) Set(value string) error {
+//	*i = append(*i, value)
+//	return nil
+//}
+
+func Main() error {
+	relativeRoot := flag.String("r", "../solo-projects/projects/gloo", "path to project absoluteRoot")
+	compileProtos := flag.Bool("gogo", true, "compile normal gogo protos")
+	flag.Parse()
+
+	absoluteRoot, err := filepath.Abs(*relativeRoot)
+	if err != nil {
+		return err
+	}
+
+	var projectDirs []string
+
+	// discover all project.json
+	if err := filepath.Walk(absoluteRoot, func(path string, info os.FileInfo, err error) error {
```

Comment:
this seems fine for now, but if we ever add javascript code to one of these projects it will get false positives since most javascript projects include `projects.json`
Instead, I think this would be better as a list or a new convention, like "solokitproject.json"

Reply:
solo-kit.json?

Reply:
agreed